src/Vectorface/SnappyRouter/Request/HttpRequest.php

```diff
@@ -145,6 +167,42 @@ public function setPost($postData)
     }
 
     /**
+     * Returns the input stream data for the current request
+     * @return string The input stream data
+     */
+    public function getInputStream()
+    {
+        // If this value has been read from the stream, retrieve it from storage
+        if (!empty($this->input[self::INPUT_METHOD_STREAM])) {
```

Comment:
This goes along with one of my comments above -> the empty string '' can be a valid body, so an empty body might be re-fetched over-and-over several times.

You could get around this by changing the default value of the `$this->input[self::INPUT_METHOD_STREAM]` to `false` or `null`.